logstash-core/src/test/java/org/logstash/ackedqueue/QueueTest.java

```diff
@@ -953,16 +955,45 @@ public void pageCapacityChangeOnExistingQueue() throws IOException {
             assertThat(q.headPage.getPageIO().getCapacity(), is(NEW_CAPACITY));
 
             // will read only within a page boundary
-            Batch b1 = q.readBatch( 10);
+            Batch b1 = q.readBatch( 10, Queue.TIMEOUT_SECOND);
             assertThat(b1.size(), is(1));
             b1.close();
 
             // will read only within a page boundary
-            Batch b2 = q.readBatch( 10);
+            Batch b2 = q.readBatch( 10, Queue.TIMEOUT_SECOND);
             assertThat(b2.size(), is(1));
             b2.close();
 
             assertThat(q.tailPages.size(), is(0));
         }
     }
+
+
+    @Test(timeout = 5000)
+    public void maximizeBatch() throws IOException, InterruptedException, ExecutionException {
+
+        // very small pages to maximize page creation
+        Settings settings = TestSettings.persistedQueueSettings(1000, dataPath);
+        try (Queue q = new Queue(settings)) {
+            q.open();
+
+            Callable<Integer> writer = () -> {
```

Comment:
nit: if you don't read the results from the execution, a `Runnable` makes that a bit more obvious

```
Runnable writer = () -> {
...
Future<?> future (or don't even bother with it since there is no need to block this test on the get)
```

Reply:
good point. Actually, I think I prefer changing it to `Callable<Void>` just because `Callable` accepts checked exceptions which makes using `Runnable` a pain for this.